proxy/hdrs/URL.cc

```diff
@@ -1158,9 +1158,51 @@ url_parse_scheme(HdrHeap *heap, URLImpl *url, const char **start, const char *en
   return PARSE_ERROR; // no non-whitespace found
 }
 
+static bool
+url_init_non_encoded_char_array(char *non_encoded_char, size_t size)
+{
+  const char *allowed_char = ":/?#[]@"                     // gen-delims
+                              "!$&'()*+,;="                // sub-delims
+                              "ABCDEFGHIJKLMNOPQRSTUVWXYZ" // ALPHA
+                              "abcdefghijklmnopqrstuvwxyz"
+                              "0123456789"                 // DIGIT
+                              "-._~"                       // other unreserved
+                              "%";                         // '%' should also not be encoded
+
+  memset(non_encoded_char, 0, size);
+  for (size_t i = 0; i < strlen(allowed_char); ++i) {
+    ink_assert((unsigned char)allowed_char[i] < size);
+    non_encoded_char[(unsigned char)allowed_char[i]] = 1;
+  }
+
+  return true;
+}
+
+static char non_encoded_char[256];
+static bool url_init_non_encoded_char_array_done = url_init_non_encoded_char_array(non_encoded_char, sizeof(non_encoded_char));
```

Comment:
Yes I see your point. I think that this properly belongs in [ParseRules.cc](https://github.com/apache/trafficserver/blob/master/lib/ts/ParseRules.cc), though all the parse bits are taken. Maybe we could reclaim something esoteric like ``is_wildmat_BIT``?